### PR TITLE
update to minimum go 1.18

### PR DESCRIPTION
**Pull request by brianonn**
## Description

updates the code to go 1.18 minimal

removed deprecated io/ioutil functions
removed deprecatd linters
use default excludes on linter

## Motivation and Context

The current code level was too low at go 1.12 

## How Has This Been Tested?

used locally 

## Types of changes

- migrate to go 1.18

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,16 +8,13 @@ default: lint test
 lint:
 	golangci-lint run \
 		--disable-all \
-		--exclude-use-default=false \
+		--exclude-use-default=true \
```
Comment (ericrpowers) [inline Makefile:11]: Let's keep it aligned with other backend repo linter settings. This should do it:
```suggestion
		--exclude-use-default=false \
		--exclude=package-comments \
```

**Review by ericrpowers — COMMENTED** [review]
Rest LGTM

File: Makefile (hunk 1 above)
Comment (brianonn) [reply Makefile:11]: souds good

**Comment by brianonn** [comment]
@ericrpowers

I committed the suggestion. please re-review. 
thanks!

**Review by ericrpowers — APPROVED** [review]

**Comment by ericrpowers** [comment]
Thanks!

**Pull request merged**